discord/state.py

```diff
@@ -1218,123 +1395,155 @@ def parse_guild_member_update(self, data) -> None:
 
         member = guild.get_member(user_id)
         if member is not None:
-            old_member = Member._copy(member)
-            member._update(data)
+            old_member = member._update(data)
             user_update = member._update_inner_user(user)
             if user_update:
                 self.dispatch('user_update', user_update[0], user_update[1])
 
-            self.dispatch('member_update', old_member, member)
+            if old_member is not None:
+                self.dispatch('member_update', old_member, member)
         else:
-            if self.member_cache_flags.joined:
+            if self.member_cache_flags.other or user_id == self.self_id or guild.chunked:
                 member = Member(data=data, guild=guild, state=self)
-
-                # Force an update on the inner user if necessary
-                user_update = member._update_inner_user(user)
-                if user_update:
-                    self.dispatch('user_update', user_update[0], user_update[1])
-
                 guild._add_member(member)
-            _log.debug('GUILD_MEMBER_UPDATE referencing an unknown member ID: %s. Discarding.', user_id)
-
-        if (presence := data.get('presence')) is not None:
-            member._presence_update(presence, tuple())
-            if old_member is not None:
-                self.dispatch('presence_update', old_member, member)
+            _log.debug('GUILD_MEMBER_UPDATE referencing an unknown member ID: %s.', user_id)
 
     def parse_guild_sync(self, data) -> None:
         print('I noticed you triggered a `GUILD_SYNC`.\nIf you want to share your secrets, please feel free to email me.')
 
-    def parse_guild_member_list_update(self, data) -> None:  # Rewrite incoming...
-        self.dispatch('raw_guild_member_list_update', data)
+    def parse_guild_member_list_update(self, data) -> None:
+        self.dispatch('raw_member_list_update', data)
         guild = self._get_guild(int(data['guild_id']))
         if guild is None:
             _log.debug('GUILD_MEMBER_LIST_UPDATE referencing an unknown guild ID: %s. Discarding.', data['guild_id'])
             return
 
-        ops = data['ops']
+        request = self._scrape_requests.get(guild.id)
 
-        if data['member_count'] > 0:
+        if (count := data['member_count']) > 0:
```

Comment:
Why is `count := data`? I couldn't find count being used anywhere in this function

Reply:
Yeah I forgot to use it. Using it now.